src/dali/dali_main.py

```diff
@@ -157,6 +157,18 @@ def _dali_main_internal(country: AbstractCountry) -> None:
             pair_converter_list.append(CcxtPairConverterPlugin(Keyword.HISTORICAL_PRICE_HIGH.value))
             LOGGER.info("No pair converter plugins found in configuration file: using default pair converters.")
 
+        pair_converters_by_cache_key: Dict[str, List[AbstractPairConverterPlugin]] = {}
+        for pair_converter in pair_converter_list:
+            cache_key = pair_converter.cache_key()
+            if cache_key not in pair_converters_by_cache_key:
+                pair_converters_by_cache_key[cache_key] = []
+            pair_converters_by_cache_key[cache_key].append(pair_converter)
+
+        for cache_key, pair_converters in pair_converters_by_cache_key.items():
+            if len(pair_converters) > 1:
+                LOGGER.error("The following pair converter plugins all attempt to store at %s: %s", cache_key, pair_converters)
+                sys.exit(1)
```

Comment:
Good sanity check! Just a nit: I think it can be simplified slightly to use only one loop:
```suggestion
        pair_converters_by_cache_key: Dict[str, AbstractPairConverterPlugin] = {}
        for pair_converter in pair_converter_list:
            cache_key = pair_converter.cache_key()
            if cache_key in pair_converters_by_cache_key:
                LOGGER.error("Internal error: the following pair converter plugins both attempt to store at %s: %s, %s", cache_key, pair_converter.name(), pair_converters_by_cache_key[cache_key].name())
                sys.exit(1)
            pair_converters_by_cache_key[cache_key] = pair_converter
```